includes/class-sensei-quiz.php

```diff
@@ -1856,14 +1856,28 @@ public static function action_buttons() {
 		$lesson_id         = Sensei()->quiz->get_lesson_id();
 		$is_quiz_completed = self::is_quiz_completed();
 		$is_reset_allowed  = self::is_reset_allowed( $lesson_id );
-		$has_actions       = $is_reset_allowed || ! $is_quiz_completed;
+		$course_id         = Sensei()->lesson->get_course_id( $lesson_id );
+		$is_learning_mode  = Sensei_Course_Theme_Option::has_learning_mode_enabled( $course_id );
+		$is_awaiting_grade = false;
+
+		$lesson_status = \Sensei_Utils::user_lesson_status( $lesson_id, get_current_user_id() );
+
+		if ( $lesson_status ) {
+			$lesson_status = is_array( $lesson_status ) ? $lesson_status[0] : $lesson_status;
+
+			$is_awaiting_grade = 'ungraded' === $lesson_status->comment_approved;
+		}
```

Comment:
Related to the previous comment, maybe we could have that method in this class being used in both places to avoid duplication? WDYT?
Feel free to disagree since I'm re-learning about Sensei and probably missing a lot of context. 🙂